test/sql/index/test_art_index.cpp

```diff
@@ -1039,6 +1040,223 @@ TEST_CASE("ART Big Range with deletions", "[art]") {
 	REQUIRE(CHECK_COLUMN(result, 0, {Value(1500)}));
 }
 
+TEST_CASE("ART Negative Range", "[art-neg]") {
+    unique_ptr<QueryResult> result;
+    DuckDB db(nullptr);
+
+    Connection con(db);
+
+    REQUIRE_NO_FAIL(con.Query("CREATE TABLE integers(i integer)"));
+    index_t n = 1000;
+    auto keys = unique_ptr<int32_t[]>(new int32_t[n]);
+    for (index_t i = 0; i < n; i++) {
+        keys[i] = i -500 ;
+    }
+
+    REQUIRE_NO_FAIL(con.Query("BEGIN TRANSACTION"));
+    for (index_t i = 0; i < n; i++) {
+        REQUIRE_NO_FAIL(con.Query("INSERT INTO integers VALUES ($1)", keys[i]));
+    }
+    REQUIRE_NO_FAIL(con.Query("COMMIT"));
+    REQUIRE_NO_FAIL(con.Query("CREATE INDEX i_index ON integers(i)"));
+
+    result = con.Query("SELECT sum(i) FROM integers WHERE i >= -500 AND i <= -498");
+    REQUIRE(CHECK_COLUMN(result, 0, {Value(-1497)}));
+    result = con.Query("SELECT sum(i) FROM integers WHERE i >= -10 AND i <= 5");
+    REQUIRE(CHECK_COLUMN(result, 0, {Value(-40)}));
+    result = con.Query("SELECT sum(i) FROM integers WHERE i >= 10 AND i <= 15");
+    REQUIRE(CHECK_COLUMN(result, 0, {Value(75)}));
+    REQUIRE_NO_FAIL(con.Query("DROP INDEX i_index"));
+    REQUIRE_NO_FAIL(con.Query("DROP TABLE integers"));
+}
+
+float generate_small_float(){
+    return static_cast <float> (rand()) / static_cast <float> (RAND_MAX);
+}
+
+float generate_float(){
+    return FLT_MIN + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(FLT_MAX-FLT_MIN)));
+}
+
+float generate_small_double(){
+    return static_cast <double> (rand()) / static_cast <double> (RAND_MAX);
+}
+
+float generate_double(){
+    return FLT_MIN + static_cast <double> (rand()) /( static_cast <double> (RAND_MAX/(FLT_MAX-FLT_MIN)));
+}
+
+double full_scan_float(float* keys, index_t size, float low, float high){
+    double sum = 0;
+    for (index_t i = 0; i < size; i ++){
+        if (keys[i] >= low && keys[i]  <= high) {
+            sum += keys[i];
+        }
+    }
+
+    return sum;
+}
+
+double full_scan_double(double* keys, index_t size, double low, double high){
+    double sum = 0;
+    for (index_t i = 0; i < size; i ++){
+        if (keys[i] >= low && keys[i]  <= high) {
+            sum += keys[i];
+        }
+    }
+
+    return sum;
+}
+
+
+TEST_CASE("ART Floating Point", "[art-float][.]") {
```

Comment:
Could you also make some smaller tests that run as part of the fast unit tests run (i.e. not allunit)?

Reply:
Can you also make a test using a float/double column as a UNIQUE or PRIMARY KEY constraint?